addons/website_quote/controllers/main.py

```diff
@@ -18,50 +15,53 @@ class sale_quote(http.Controller):
     def view(self, order_id, pdf=None, token=None, message=False, **post):
         # use SUPERUSER_ID allow to access/view order for public user
         # only if he knows the private token
-        order = request.registry.get('sale.order').browse(request.cr, token and SUPERUSER_ID or request.uid, order_id, request.context)
-        now = time.strftime('%Y-%m-%d')
-        dummy, action = request.registry.get('ir.model.data').get_object_reference(request.cr, request.uid, 'sale', 'action_quotations')
+        SaleOrder = request.env['sale.order']
+        if token:
+            order = SaleOrder.sudo().browse(order_id)
```

Comment:
why not search based on order_id + token ? instead read and verify at line no 26